Helper/ProductIndexer.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Aligent\PrerenderIo\Helper;
+
+use Magento\Catalog\Model\ProductFactory;
+use Magento\ConfigurableProduct\Model\Product\Type\Configurable;
+
+/**
+ * ProductIndexer helper class
+ */
+class ProductIndexer
+{
+    /**
+     * @var Configurable
+     */
+    private Configurable $configurable;
+
+    /**
+     * @var ProductFactory
+     */
+    private ProductFactory $productFactory;
+
+    /**
+     * @param Configurable $configurable
+     * @param ProductFactory $productFactory
+     */
+    public function __construct(
+        Configurable $configurable,
+        ProductFactory $productFactory
+    ) {
+        $this->configurable = $configurable;
+        $this->productFactory = $productFactory;
+    }
+
+    /**
+     * Returns parent entity id(s)
+     *
+     * @param $simpleProductId
+     * @return array
+     */
+    public function getParentEntityId($simpleProductId): array
+    {
+        $parentIds = [];
+        $simpleProduct = $this->productFactory->create()->load($simpleProductId);
+        if ($simpleProduct->getTypeId() == 'simple') {
```

Comment:
Hi @mausham-shrestha-aligent, 

It would be better to check the visibility of the simple product as well.

Reply:
@ryanjehan-aligent - just to confirm, are you suggesting something like this?

 ```
if ($simpleProduct->getTypeId() 
      && $simpleProduct->getVisibility() != Visibility::VISIBILITY_NOT_VISIBLE 
) {
....
...
}
```

Reply:
@ryanjehan-aligent do you think we should still load product and check for its "type" and "visibility"  if we do it like this below? 
```
$parentIds = array_unique($this->configurable->getParentIdsByChild($entityIds));
$entityIds = array_merge($entityIds, $parentIds);
```
I think doing this will still allow the simple product to be in the list irrespective of whether it is attached to the configurable product on not?

cc @aligent-lturner

Reply:
@mausham-shrestha-aligent I wouldn't - the simple product's visibility shouldn't really matter - you'd want to recache the parent anyway

Reply:
And yes, getting all the parent ids and merging should be all you need